### ISSUE TITLE
BUG: Improve baseline update in pipeline

### ISSUE THREAD

**Issue by markusguenther** [neos/flow-development-collection#3022]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

In the github workflow we have the following two steps.

```
      - name: Static analysis
        if: matrix.static-analysis == 'psalm'
        run: composer test-static

      - name: Update psalm baseline
        if: ${{ failure() && github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged == true }}
        run: |
          composer psalm-baseline-update
          cd ./Packages/Framework
          git add psalm-baseline.xml
          git -c user.name='gh-action' -c user.email='gh-action@noreply.github.com' commit -m "TASK: Update psalm-baseline"
          git push -f origin HEAD:${FLOW_TARGET_VERSION}
```

Sadly, the plan that we update the baseline automatically when the static analysis fails, is not working.
The Update psalm baseline is not running when we have an error on the analysis before and the update itself can not update the base because the git ignore file is preventing the update and a commit is also not permitted.

So this needs a complete overhaul.

### Expected Behavior

The baseline update works after a failure.

### Steps To Reproduce

You need a failing psalm run.

### Environment

```markdown
- Flow:
- PHP:
```

**Comment by kitsunet** [comment]
Ahhh, that is nasty

**Comment by markusguenther** [comment]
Yup ... for the error, we just need to add `continue-on-error: true` to the step. But for the whole change detection via git and committing the change, we need to adjust the whole baseline step.

**Comment by markusguenther** [comment]
**Idea**

* Install the `flow-development-distribution` with sources so that we have the own git repository inside the Framework Folder
* If the error for git add still be there we can add a force
* And then make the git commit work (currently we have no permission)
